src/plugins/intel_gpu/tests/functional/shared_tests_instances/single_layer_tests/mat_mul.cpp

```diff
@@ -10,6 +10,27 @@ namespace {
 using ov::test::MatMulLayerTest;
 using ov::test::utils::InputLayerType;
 
+class GPUMatMulLayerTest : public MatMulLayerTest {
+protected:
+    void SetUp() override {
+        MatMulLayerTest::SetUp();
+        ov::test::MatMulLayerTestParamsSet params = GetParam();
+        std::vector<ov::test::InputShape> shapes;            // Input Shapes
+        std::pair<bool, bool> transpose_type;              // Transpose inputs
+        ov::element::Type model_type;                  // Model type
+        ov::test::utils::InputLayerType input_layer_type;    // Secondary input type
+        std::map<std::string, std::string>  additional_config; // Additional network configuration
+        std::tie(shapes, transpose_type, model_type, input_layer_type, targetDevice, additional_config) = this->GetParam();
+        // Some rounding float to integer types on GPU may differ from CPU, and as result,
+        // the actual values may differ from reference ones on 1 when the float is very close to an integer,
+        // e.g 6,0000023 calculated on CPU may be cast to 5 by OpenCL convert_uchar function.
+        // That is why the threshold is set 1.f for integer types.
+        if (targetDevice == "GPU" && model_type == ov::element::f16) {
+            rel_threshold = 0.005;
+        }
+    }
+};
+
```

Comment:
I think test should be implemented from unittest with enforcing kernel instance.

Reply:
Added unit test for the purpose. Thanks!